### ISSUE TITLE
Google Play Music: empty Playlists

### ISSUE THREAD

**Issue by JohannesKuhaupt** [vincelwt/harmony#84]
I only use Harmony with Google Play Music. All Google Music playlists appear on the left sidepane. So far so good. But not all playlists are loaded correct. The "thumbs up" playlist seems correct and contains all songs. But other playlists are empty or songs are missing. So these playlists don't contain all the songs, which is very annoying. The missing songs appear in the global Google Music Playlist or in the "thumbs up" playlist. This problem affects normal playlists and offline playlists.

System: Linux Mint 18 XFCE / Ubuntu 16.04 Base / amd64
Harmony 0.3.2

A fix for this problem would be very nice ;-)

btw: Harmony is a great music app with a very clean interface!

Thanks, 
Johannes

**Comment by vincelwt** [comment]
Hi!
Are you using Google Play Music all access ?
Do find something in common with the disappearing tracks ? Maybe only all-access tracks are disappearing ? That would be a great help.

**Comment by JohannesKuhaupt** [comment]
Hi,
I don't know what you mean with "all-access tracks". I'am using the Google Play Music Flatrate (I pay every month) and uploaded some songs, which I haven't found on Google Music. The disappeared tracks / missing tracks are both types. Some uploaded tracks are missing and some Google Music tracks are missing. I can't figure out any similarities / things that the missing tracks have in common. Sorry.

And I have to correct my frist bug post. The Global Google Music Playlist has also missing tracks. So I think, that there is a global missing track problem with Google Play Music.

**Comment by vincelwt** [comment]
Mmh, could you try opening developer tools (press Ctrl+Alt+i), go to the console tab and copy paste everything ? See if an error popped up

**Comment by ManuelPimienta** [comment]
Hello, I try to login on Google Play Music and failed to start session, and now can not close the setting manager because asks me for the Google account. Sorry for my english, and thanks for the software.

**Comment by ManuelPimienta** [comment]
![error_harmony](https://cloud.githubusercontent.com/assets/20049211/19418663/ee9732b6-938d-11e6-9776-6cf1861b91d4.png)
Hello, I try to login on Google Play Music and failed to start session, and now can not close the setting manager because asks me for the Google account. Sorry for my english, and thanks for the software.

**Comment by vincelwt** [comment]
@ManuelPimienta Try resetting all the settings and data (at the bottom of settings page)

**Comment by JohannesKuhaupt** [comment]
Sorry I am on tour. I installed Harmony on my notebook and got the same error / problem.

There is one error message: "Uncaught TypeError: Cannot read property 'playlists' of undefined (utils.js:74)"

**Comment by ManuelPimienta** [comment]
@vincelwt It's works, thanks.

**Comment by vincelwt** [comment]
@JohannesKuhaupt I don't think this is the error causing the problem. Could you copy paste the whole ouput, because some errors are printed like infos but are still errors (I need to correct that)

**Comment by JohannesKuhaupt** [comment]
This is the console output:

We are on a -linux- system
Mediakeys module not found.
changed ObjectIdentity: "Harmony"__proto__: Object
changed Object
changed Object
local fetched, testing internet
Testing internet...
listView
Latest release is v0.3.2 and we're running undefined
Everything over
listView
listView

When I select one empty playlist (which is usually not empty), there is this error:
player.js:86 Uncaught TypeError: Cannot read property 'title' of undefinedplayTrack @ player.js:86playByIndex @ main.js:6ondblclick @ index.html:1
2main.js:411 listView

**Comment by vincelwt** [comment]
The error you're reporting is because when you double clicked on the playlist, as there is no tracks in this playlist, it can't play the first track. 
But that's not the why of the problem, so it must be something with the Google Music API

**Comment by JohannesKuhaupt** [comment]
Sorry, I can't produce other error messages

**Comment by arvigeus** [comment]
I had the same problem when tried Harmony 0.2. Will watch this thread for development.

**Comment by JohannesKuhaupt** [comment]
Tried Harmony 0.44a (current version, with config reset) and still got the same problem.

**Comment by nypher** [comment]
0.4.4 - Only 3 playlists are loading content on Windows, the others (I have 30 playlists) don't show anything.

**Comment by vincelwt** [comment]
Should be fixed with next release (0.5.1) :)

**Comment by vincelwt** [comment]
0.5.1 released, could you try to confirm?

**Comment by JohannesKuhaupt** [comment]
I tried 0.5.1 and it seems way better. Many missing songs are now visible in playlists. But there are still some empty playlists. Sorry :-/ But it is a big step forward!

**Comment by vincelwt** [comment]
@JohannesKuhaupt Noted. How large is your library? Would you say more than 40k tracks?

**Comment by JohannesKuhaupt** [comment]
About 9-10k tracks,

**Comment by vincelwt** [comment]
Are this time, the missing tracks, song you bought? Or is there also missing songs you uploaded?

**Comment by JohannesKuhaupt** [comment]
Both!

**Comment by vincelwt** [comment]
@ericfernandez I may have have fixed this in next release, I'll let you know when available
